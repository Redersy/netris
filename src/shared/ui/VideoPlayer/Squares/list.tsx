import { SquaresListType } from "../model";

export const List = ({ list }: SquaresListType) => (
  <>
    {list.map((item) => (
      <div className="square" key={`square-${item.id}`} data-testid={`square-${item.id}`} style={{ ...item.zone }} />
    ))}
  </>
);
