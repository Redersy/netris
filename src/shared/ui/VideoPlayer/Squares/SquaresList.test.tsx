import React from "react";
import { render, screen } from "@testing-library/react";
import { List } from "./list";

const list = [
  {
    id: 1,
    zone: {
      width: 10,
      height: 10,
      left: 0,
      top: 0
    },
    start: 1,
    finish: 100
  }
];
describe("Squares", () => {
  test("Render list", () => {
    render(<List list={list} />);

    const square = screen.getByTestId("square-1");
    expect(square).toBeInTheDocument();
  });
  test("Snapshot list", () => {
    const { asFragment } = render(<List list={list} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
