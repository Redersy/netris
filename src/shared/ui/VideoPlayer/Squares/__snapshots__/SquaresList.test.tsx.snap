// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squares Snapshot list 1`] = `
<DocumentFragment>
  <div
    class="square"
    data-testid="square-1"
    style="width: 10px; height: 10px; left: 0px; top: 0px;"
  />
</DocumentFragment>
`;
